src/user/service.ts

```diff
@@ -159,70 +156,47 @@ export async function findUserInfoByUser(user: User): Promise<UserInfo> {
  * @param userInfo - new UserInfo object to update the UserInfo record with.
  * @returns The updated UserInfo record.
  */
-export async function updateUserInfo(user: User, userInfo?: UserInfo): Promise<void> {
-
-  if (!userInfo) return;
+export async function updateUserInfo(user: User, userInfo: UserInfo): Promise<void> {
 
-  const data = {
-    principal_id: user.id,
+  const newUserInfoValues: Partial<UserInfoRecord> = {
     name: userInfo.name,
     locale: userInfo.locale,
     given_name: userInfo.givenName,
     family_name: userInfo.familyName,
-    birthdate: userInfo.birthDate,
-    address: userInfo.address ? JSON.stringify(userInfo.address) : null,
-    zoneinfo: userInfo.zoneInfo
+    birthdate: userInfo.birthdate,
+    address: userInfo.address ? JSON.stringify(userInfo.address.streetAddress) : null,
+    zoneinfo: userInfo.zoneinfo
   };
 
-  const result = await db('user_info').where({ principal_id: user.id }).update(data);
-
-  // No rows for existing user_info was found, so insert a new record
-  if (result === 0) {
-    await db('user_info')
-      .insert({
-        ...data,
-        created_at: Date.now(),
-        modified_at: Date.now(),
-      })
-      .onConflict('principal_id')
-      .merge({
-        name: data.name,
-        locale: data.locale,
-        given_name: data.given_name,
-        family_name: data.family_name,
-        birthdate: data.birthdate,
-        address: data.address,
-        zoneinfo: data.zoneinfo,
-        modified_at: Date.now(),
-      });
-  }
+  await db('user_info')
+    .insert({
+      ...newUserInfoValues,
+      principal_id: user.id,
+      created_at: Date.now(),
+      modified_at: Date.now(),
+    })
+    .onConflict('principal_id')
+    .merge({
+      ...newUserInfoValues,
+      modified_at: Date.now(),
+    });
 }
 
 /**
- * @description - Given a UserInfo record from database, converts the record to a UserInfo object.
- * @param record - The UserInfo record to convert to a UserInfo object.
- * @returns The UserInfo object.
+ * Given a UserInfo record from database, converts the record to a UserInfo object.
+ * @param {UserInfoRecord} record - The UserInfo record to convert to a UserInfo object.
  */
 export async function recordToModel(record: UserInfoRecord): Promise<UserInfo> {
 
   return {
-    name: record.name || null,
-    locale: record.locale || null,
-    givenName: record.given_name || null,
-    familyName: record.family_name || null,
-    birthDate: record.birthdate || null,
+    name: record.name,
+    locale: record.locale,
+    givenName: record.given_name,
+    middleName: record.middle_name,
+    familyName: record.family_name,
+    birthdate: record.birthdate,
     address: record.address ? JSON.parse(record.address) : null,
-    zoneInfo: record.zoneinfo || null,
-    modifiedAt: record.modified_at ? new Date(+record.modified_at) : null,
+    zoneinfo: record.zoneinfo,
+    metadata: record.metadata ? JSON.parse(record.metadata) : {},
```

Comment:
At this point, I assume if none of the values exist, theyd already be `null` due to the database response so we dont have to fallback or `??`

Reply:
Yes, that's why I removed most of the `??` here. I kept the extra check for metadata just in case someone does something funky with that field and tries to clear it by setting it to an empty string or null. I guess I don't 100% trust that there's valid JSON in there but probably that's silly?